be/src/udf/udf.h

```diff
@@ -786,6 +792,31 @@ struct HllVal : public StringVal {
     void agg_merge(const HllVal &other);
 };
 
+struct CollectionVal : public AnyVal {
```

Comment:
Actually, if this struct is defined, the struct can not be changed, because some UDF function will use this struct. If this is only used for array, we can make it ArrayVal?

Reply:
I referenced the implementation of Impala,  nested type(array, map)  is used same struct to describe, the different is the data in struct, so I think we can also use same struct. Personally, I also think called ArrayVal and ArrayValue is good idea